base/util_test.go

```diff
@@ -553,32 +554,52 @@ func TestRedactBasicAuthURL(t *testing.T) {
 		},
 		{
 			input:    "http://username:password@hostname",
-			expected: "http://****:****@hostname",
+			expected: "http://xxxxx:xxxxx@hostname",
 		},
 		{
 			input:    "https://username:password@example.org:8123",
-			expected: "https://****:****@example.org:8123",
+			expected: "https://xxxxx:xxxxx@example.org:8123",
 		},
 		{
 			input:    "https://username:password@example.org/path",
-			expected: "https://****:****@example.org/path",
+			expected: "https://xxxxx:xxxxx@example.org/path",
 		},
 		{
 			input:    "https://username:password@example.org:8123/path?key=val&email=me@example.org",
-			expected: "https://****:****@example.org:8123/path?key=val&email=me@example.org",
+			expected: "https://xxxxx:xxxxx@example.org:8123/path?key=val&email=me@example.org",
 		},
 		{
 			input:    "https://foo%40bar.baz:my-%24ecret-p%40%25%24w0rd@example.com:8888/bar",
-			expected: "https://****:****@example.com:8888/bar",
+			expected: "https://xxxxx:xxxxx@example.com:8888/bar",
 		},
 		{
 			input:    "https://example.com/does-not-count-as-url-embedded:basic-auth-credentials@qux",
 			expected: "https://example.com/does-not-count-as-url-embedded:basic-auth-credentials@qux",
 		},
+		{
+			input:    "http://example.org",
+			expected: "http://example.org",
+		},
+		{
+			input:    "http://example.org:1234",
+			expected: "http://example.org:1234",
+		},
+		{
+			input:    "http://foo:bar@example.org",
+			expected: "http://xxxxx:xxxxx@example.org",
+		},
+		{
+			input:    "http://foo:bar@example.org:1234",
+			expected: "http://xxxxx:xxxxx@example.org:1234",
+		},
+		{
+			input:    "http://foo:p@ssw0rd@example.org",
+			expected: "http://xxxxx:xxxxx@example.org",
+		},
 	}
 
 	for _, test := range tests {
-		goassert.Equals(t, RedactBasicAuthURL(test.input), test.expected)
+		goassert.Equals(t, RedactBasicAuthURLUserAndPassword(test.input), test.expected)
```

Comment:
Seems like a good time to get off the old assert library and use subtests whilst we have all the test cases already in a table.

```suggestion
		t.Run(test.input, func(t *testing.T) {
			assert.Equal(t, test.expected, RedactBasicAuthURLUserAndPassword(test.input))
		})
```